lib/trino-filesystem/src/main/java/io/trino/filesystem/TrinoFileSystem.java

```diff
@@ -228,6 +229,18 @@ Set<Location> listDirectories(Location location)
     Optional<Location> createTemporaryDirectory(Location targetPath, String temporaryPrefix, String relativePrefix)
             throws IOException;
 
+    /**
+     * Returns the direct pre-signed URI location for the given storage location.
+     *
+     * This is used to retrieve the file directly from the storage location without going through
+     * the Trino server. This is useful for large files where the Trino server would be a bottleneck.
+     */
+    default Optional<UriLocation> preSignedUri(Location location, Duration ttl)
+            throws IOException
+    {
+        return Optional.empty();
```

Comment:
What does return of Optional.empty() mean? Please add info to the Javadoc. 
Is it "not supported" flag? I would prefer to throw exception in that case, and have some separate method from which you may get filesystem "capabilities".